src/utils/serve-functions.js

```diff
@@ -515,21 +515,35 @@ const setupFunctionsBuilder = async ({ config, errorExit, functionsDirectory, lo
 
   log(`${NETLIFYDEVLOG} Function builder ${chalk.yellow(functionBuilder.builderName)} detected${npmScriptString}.`)
 
-  const debouncedBuild = debounce(getBuildFunction({ functionBuilder, log }), DEBOUNCE_WAIT, {
-    leading: true,
-    trailing: true,
-  })
+  const cache = new Map()
+
+  const buildFunction = getBuildFunction({ functionBuilder, log })
+
+  // `memoizedBuild` will avoid consecutive calls to the build function with
+  // the same arguments until the previous invokation has finished running.
+  const memoizedBuild = (path, eventType) => {
+    const key = `${eventType}@${path}`
+
+    if (!cache.has(key)) {
+      cache.set(
+        key,
+        buildFunction(path, eventType).finally(() => cache.delete(key)),
+      )
+    }
+
+    return cache.get(key)
```

Comment:
I couldn't find where the return value is used

Reply:
It's not used currently, but `debouncedBuild` preserved the return value of the build function and I thought `memoizedBuild` should respect that.